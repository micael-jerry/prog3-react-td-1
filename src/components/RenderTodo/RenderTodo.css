.list {
    background-color: #f66969;
    border-radius: 10px;
}
