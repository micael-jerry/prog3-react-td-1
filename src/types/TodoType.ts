export interface TodoType {
    title: string;
    description: string;
    completed: boolean;
}
