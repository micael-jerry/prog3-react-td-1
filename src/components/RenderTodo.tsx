import React from "react";
import { TodoType } from "../types/TodoType";

const RenderTodo: React.FC<{
	todoList: TodoType[];
	checkList: (todoList: any) => any;
}> = (props) => {
	const { todoList, checkList } = props;

	return (
		<>
			<div>
				<h1>TODO</h1>
				<ul>
					{todoList.map((t, i) => {
						if (!t.completed) {
							return (
								<>
									<input
										key={t.description.trim()}
										type={"checkbox"}
										name="check"
										id={i + ""}
										onClick={checkList}
									/>
									<li key={(i + t.description).trim()}>
										{t.title}
									</li>
								</>
							);
						} else {
							return <></>;
						}
					})}
				</ul>
			</div>
			<div>
				<h1>DONE</h1>
				<ul>
					{todoList.map((t, i) => {
						if (t.completed) {
							return (
								<li key={(t.description + i).trim()}>
									{t.title}
								</li>
							);
						} else {
							return <></>;
						}
					})}
				</ul>
			</div>
		</>
	);
};

export default RenderTodo;
