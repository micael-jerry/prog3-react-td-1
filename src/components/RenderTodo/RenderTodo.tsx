import React from "react";
import { TodoType } from "../../types/TodoType";
import "./RenderTodo.css";

const RenderTodo: React.FC<{
    todoList: TodoType[];
    checkList: (todoList: any) => any;
}> = (props) => {
    const { todoList, checkList } = props;

    return (
        <div className="row g-4 py-5 row-cols-1 row-cols-lg-2">
            <div>
                <div className={"list p-4"}>
                    <h2>TODO</h2>
                    <ul
                        data-testid={"list-group-todo"}
                        className={"list-group"}
                    >
                        {todoList.map((t, i) => {
                            if (!t.completed) {
                                return (
                                    <>
                                        <li
                                            className={"list-group-item"}
                                            key={(
                                                i +
                                                t.description +
                                                t.title
                                            ).trim()}
                                        >
                                            <div className="row align-items-center">
                                                <div
                                                    className={
                                                        "col-lg-2 col-md-2 col-sm-2"
                                                    }
                                                >
                                                    <input
                                                        type={"checkbox"}
                                                        key={t.description.trim()}
                                                        id={i.toString()}
                                                        onClick={checkList}
                                                    />
                                                </div>
                                                <div
                                                    className={
                                                        "col-lg-10 col-md-10 col-sm-10"
                                                    }
                                                >
                                                    <h4>{t.title}</h4>
                                                    <p>{t.description}</p>
                                                </div>
                                            </div>
                                        </li>
                                    </>
                                );
                            } else {
                                return <></>;
                            }
                        })}
                    </ul>
                </div>
            </div>
            <div>
                <div className={"list p-4"}>
                    <h2>DONE</h2>
                    <ul
                        data-testid={"list-group-done"}
                        className={"list-group"}
                    >
                        {todoList.map((t, i) => {
                            if (t.completed) {
                                return (
                                    <li
                                        className={"list-group-item"}
                                        key={(t.description + i).trim()}
                                    >
                                        <h4>{t.title}</h4>
                                        <p>{t.description}</p>
                                    </li>
                                );
                            } else {
                                return <></>;
                            }
                        })}
                    </ul>
                </div>
            </div>
        </div>
    );
};

export default RenderTodo;
