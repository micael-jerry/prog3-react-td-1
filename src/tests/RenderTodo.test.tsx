import { fireEvent, render, screen } from "@testing-library/react";
import React from "react";
import RenderTodo from "../components/RenderTodo/RenderTodo";
import { TodoType } from "../types/TodoType";

const todo1: TodoType = {
    title: "todo1",
    description: "description1",
    completed: false,
};

const todo2: TodoType = {
    title: "todo2",
    description: "description2",
    completed: false,
};

const todo3: TodoType = {
    title: "todo3",
    description: "description3",
    completed: true,
};

const todoListMock: TodoType[] = [todo1, todo2, todo3];

test("title TODO", () => {
    const checkListMock = jest.fn();
    render(<RenderTodo todoList={[]} checkList={checkListMock} />);
    const titleTodo = screen.getByText("TODO");

    expect(titleTodo).toBeInTheDocument();
});

test("title DONE", () => {
    const checkListMock = jest.fn();
    render(<RenderTodo todoList={[]} checkList={checkListMock} />);
    const titleTodo = screen.getByText("DONE");

    expect(titleTodo).toBeInTheDocument();
});

test("Render list to todo", () => {
    const todoList: TodoType[] = [...todoListMock];
    const checkListMock = jest.fn();
    render(<RenderTodo todoList={todoList} checkList={checkListMock} />);
    const listTodoElement: HTMLElement = screen.getByTestId("list-group-todo");

    expect(listTodoElement.childNodes.length).toBe(2);
});

test("Render list to done", () => {
    const todoList: TodoType[] = [...todoListMock];
    const checkListMock = jest.fn();
    render(<RenderTodo todoList={todoList} checkList={checkListMock} />);
    const listDoneElement: HTMLElement = screen.getByTestId("list-group-done");

    expect(listDoneElement.childNodes.length).toBe(1);
});

test("Click on the checkbox to change the status of a todo to done", () => {
    const todoList: TodoType[] = [...todoListMock];
    const checkListMock = jest.fn();
    render(<RenderTodo todoList={todoList} checkList={checkListMock} />);
    const listDoneElement: HTMLElement[] = screen.getAllByTestId("check-todo");

    fireEvent.click(listDoneElement[0]);
    expect(checkListMock.mock.calls.length).toBe(1);
});

test("[SNAPSHOT] test render RenderTodo", () => {
    const checkListMock = jest.fn();
    const { container } = render(
        <RenderTodo todoList={[]} checkList={checkListMock} />
    );

    expect(container).toMatchSnapshot();
});
