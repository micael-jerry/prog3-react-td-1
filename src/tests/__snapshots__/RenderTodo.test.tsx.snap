// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[SNAPSHOT] test render RenderTodo 1`] = `
<div>
  <div
    class="row g-4 py-5 row-cols-1 row-cols-lg-2"
  >
    <div>
      <div
        class="list p-4"
      >
        <h2>
          TODO
        </h2>
        <ul
          class="list-group"
          data-testid="list-group-todo"
        />
      </div>
    </div>
    <div>
      <div
        class="list p-4"
      >
        <h2>
          DONE
        </h2>
        <ul
          class="list-group"
          data-testid="list-group-done"
        />
      </div>
    </div>
  </div>
</div>
`;
